pkg/resource/typed/typed_resource.go

```diff
@@ -62,8 +61,10 @@ func (t *Resource[T, RD]) ResourceDefinition() spec.ResourceDefinitionSpec {
 //
 // UnmarshalProto requires that the spec implements the protobuf.ProtoUnmarshaller interface.
 func (t *Resource[T, RD]) UnmarshalProto(md *resource.Metadata, protoBytes []byte) error {
-	// Go doesn't allow to do type assertion on a generic type T, so use reflection instead.
-	protoSpec, ok := reflect.ValueOf(&t.spec).Interface().(protobuf.ProtoUnmarshaler)
+	var spec any = &t.spec
+
+	// Go doesn't allow to do type assertion on a generic type T, so use intermediate any value.
+	protoSpec, ok := spec.(protobuf.ProtoUnmarshaler)
```

Comment:
```suggestion
	protoSpec, ok := any(&t.spec).(protobuf.ProtoUnmarshaler)
```

This should do, and probably is idiomatic.